### PR TITLE
Update build dependencies and Gulpfile

### PR DESCRIPTION
**Pull request by rosswintle**
I got a bunch of dependencies after merging #10.

This PR updates all of the build tooling and the Gulpfile.

## Testing

1. Pull the changes
2. Run each of the gulp tasks
3. Ensure that the CSS and JS output hasn't changed (except for variable names and whitespace)

### CODE REVIEW

**Comment by rosswintle** [comment]
@sanjucta 

As you reviewed #10 would you mind quickly reviewing this as well please?

Thanks

**Comment by rosswintle** [comment]
Hi @sanjucta.

I've re-updated this as it has [a load of dependabots](https://github.com/deliciousbrains/wp-post-promoter/security/dependabot)

I've not tested it but the changes seem OK.

Can you review this please? Then we can merge it and get our sites that use it updated.

Thanks

File: Gulpfile.js [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
-gulp.task( 'default', [ 'css', 'js' ] );
-
-gulp.task( 'watch', [ 'css', 'js' ], function() {
+function runLivereload() {
 	livereload.listen();
-	gulp.watch( [ 'assets/scss/**/*.scss' ], [ 'css' ] );
-	gulp.watch( [ 'assets/js/**/*.js' ], [ 'js' ] );
-} );
\ No newline at end of file
+	gulp.watch( [ 'assets/scss/**/*.scss' ], css );
```
Comment (tstokes8040) [inline Gulpfile.js:34]: @rosswintle this is failing. I believe this needs to be changed from `css` to `sass`.

**Review by tstokes8040 — CHANGES_REQUESTED** [review]

File: Gulpfile.js (hunk 1 above)
Comment (tstokes8040) [reply Gulpfile.js:34]: I was wrong. That should be `compileSass`. Then you should be good.

Comment (rosswintle) [reply Gulpfile.js:34]: I made a better fix. And the `watch` job now works properly too. Thanks!

**Comment by rosswintle** [comment]
@tstokes8040 Could you do another run-through of this. Should all be working now. Cheers!

**Review by tstokes8040 — APPROVED** [review]
Looks great! I like the change on the watch call!

**Comment by tstokes8040** [comment]
@rosswintle All set!

**Comment by rosswintle** [comment]
Let's see what Dependabot thinks then...

**Pull request merged**